### ISSUE TITLE
Client still crashes on disguise (party window)

### ISSUE THREAD

**Issue by vBrenth** [HerculesWS/Hercules#2393]
**Describe the bug**
Client still crashes when party member are in Disguise and you check party-windows.

**To Reproduce**
1. 2 players on party.
2. Player A: @disguise Poring
3. Player B: While Alt+Z, Walk far on the screen without seeing player A, then go Back then do Alt+Z twice.

**Expected behavior**
I think its the HP bar that makes the client crash.

**System specs (please complete the following information):**
 - OS: Windows 10
 - Hercules Version: Latest
 - Mode: Pre-renewal
 - Packet version: 20150916
 - Client type: RE

**Additional context**
It is related to https://github.com/HerculesWS/Hercules/commit/3e6e76b8c673e99d240a700912eb4d4a68b3be30 But its still bug.

**Comment by AnnieRuru** [comment]
`    * Packet version: 20150916`
test on 20180621 doesn't crash
however I do get this 0/0 when player suddenly undisguise ... though
![](https://raw.githubusercontent.com/AnnieRuru/customs/master/screenshots/screen2019Hercules033.jpg)

**Comment by vBrenth** [comment]
Yep, I think its the HP counter because i got 6871232175/** number then re-opening will crash it.

**Comment by EyesOfAHawk** [comment]
Confirmed crash using 20160113.

**Comment by AnnieRuru** [comment]
somewhat related https://github.com/HerculesWS/Hercules/issues/2104
from what I understand is that we don't send the location of the player when player is disguise
this was done long ago to prevent client crash

and now this sending hp bar ... is sent to every party member in the area ....
some information mismatch ....

**Comment by guilherme-gm** [comment]
I did some tests (using 2018-10-02) and **my guess** is that the problem is the objectType we send to clients. If you disguise as a player class, it works fine, but once you disguise as a npc, the problem shows up.

disguising calls `clif_spawn` which calls `clif_spawn_unit`. If you change `p.objecttype = clif->bl_type(bl);` to always send the object as a "player", the HP bar and party window keeps working, but everybody will see your character as a novice (probably because the client can't find the monster ID as a player job, so it uses the default sprite)

The HP bar going to 0 when you undisguise seems to be fixable if you call `clif->party_hp` after `clif->spawn` in `pc->disguise`. But this doesn't fix the real problem.
